src/lib/mirage_block_ocaml.ml

```diff
@@ -360,16 +371,23 @@ let process_one t =
             return Response.Delete
           | Request.Flush h ->
             let t = Handle.find_or_quit h in
-            Block.flush t.Handle.base
-            >>= fun () ->
-            return Response.Flush
+            let open Lwt.Infix in
+            (* Block.write_error <> Qcow.write_error *)
+            begin Block.flush t.Handle.base
+              >>= function
+              | Error `Disconnected -> Lwt.return (Error `Disconnected)
+              | Error `Is_read_only -> Lwt.return (Error `Is_read_only)
```

Comment:
But yes, the type of error should probably be made explicit in https://github.com/mirage/ocaml-qcow/blob/master/lib/qcow.mli#L21 (e.g. using `with type write_error := [ Block.write_error | `Msg of string]`